core/deadline.go

```diff
@@ -207,18 +206,32 @@ func NewDutyDeadlineFunc(ctx context.Context, eth2Cl eth2wrap.Client) (DeadlineF
 	}
 
 	return func(duty Duty) (time.Time, bool) {
-		if duty.Type == DutyExit || duty.Type == DutyBuilderRegistration {
+		switch duty.Type {
+		case DutyExit, DutyBuilderRegistration:
 			// Do not timeout exit or registration duties.
 			return time.Time{}, false
+		default:
 		}
 
-		start := genesisTime.Add(slotDuration * time.Duration(duty.Slot))
-		delta := slotDuration * time.Duration(lateFactor)
-		if delta < lateMin {
-			delta = lateMin
+		var (
+			start    = genesisTime.Add(slotDuration * time.Duration(duty.Slot))
+			margin   = slotDuration / marginFactor
+			duration time.Duration
+		)
+
+		switch duty.Type {
+		case DutyProposer:
+			duration = slotDuration / 3
+		case DutyAttester, DutyAggregator, DutyPrepareAggregator:
+			duration = 2 * slotDuration
+		case DutySyncMessage:
+			duration = 2 * slotDuration / 3
+		case DutyRandao:
+			duration = 2 * slotDuration
+		default:
+			duration = slotDuration
 		}
```

Comment:
A total nitpick from my side here:
Probably we can be a bit more explicit with all the duties we usually keep track of, instead of putting them in the `default`. Also, add some comments and make it a bit more closer to natural language (2 / 3 * slot, instead of 2 * slot / 3) .

```suggestion
		switch duty.Type {
		case DutyProposer:
			duration = 1 / 3 * slotDuration
		case DutyAttester, DutyAggregator, DutyPrepareAggregator:
			// Even though attestations and aggregations are acceptable even after 2 slots, the rewards are heavily diminished.
			duration = 2 * slotDuration
		case DutySyncMessage:
			duration = 2 / 3 * slotDuration
		case DutySyncContribution, DutyPrepareSyncContribution:
			duration = slotDuration
		case DutyRandao:
			// Randao should be accepted as long as it is not after the proposer's slot.
			// However, given how cheap the operation is, the beacon node failing to provide it for 2 slots signals that there is something wrong with the beacon node itself.
			duration = 2 * slotDuration
		default:
			duration = slotDuration
		}
```